build_container.sh

```diff
@@ -17,6 +17,12 @@ DEBIAN_FRONTEND="noninteractive" apt-get install --no-install-recommends -y \
 # cleanup
 apt-get clean && rm -rf /var/lib/apt/lists/*
 
+# help musl-gcc find linux headers
```

Comment:
I'd like to know why we need this workaround, is it something specific for this image (Ubuntu:22.04)?

Reply:
Without it compiling with musl-gcc fails like this: https://buildkite.com/rust-vmm/vhost-ci/builds/615#018cd414-a613-4e9f-804f-c698d683cf64
From what I understand, when compiling, `musl-gcc` will only look at the `musl` specific headers (in `/usr/include/$ARCH-linux-musl`) and will not find default linux headers. So we need to manually link linux headers to the `musl` directory.

Reply:
Yes, but my question is why doesn't the distro do that already?
Is it a bug? Or do they expect us to have to configure `musl-gcc` differently?

Also, are we sure libraries/executables compiled this way with `musl` will work?

Reply:
> Yes, but my question is why doesn't the distro do that already?

I would like to know the answer to this question myself.

> Also, are we sure libraries/executables compiled this way with musl will work?

Considering we just point the compiler to the system headers, I don't see why would final lib/exec will not work. Also this technique is used in FC: https://github.com/firecracker-microvm/firecracker/blob/ab3e4fe129225c9620bd6591ea8ab3d0f03337f0/tools/devctr/Dockerfile#L141 and it never caused any problems.

Reply:
> > Yes, but my question is why doesn't the distro do that already?
> 
> I would like to know the answer to this question myself.

eheh :-)

> 
> > Also, are we sure libraries/executables compiled this way with musl will work?
> 
> Considering we just point the compiler to the system headers, I don't see why would final lib/exec will not work. Also this technique is used in FC: https://github.com/firecracker-microvm/firecracker/blob/ab3e4fe129225c9620bd6591ea8ab3d0f03337f0/tools/devctr/Dockerfile#L141 and it never caused any problems.

Okay, I was just looking for confirmation on that. Next time I suggest explaining better (at least in the commit description), how it was tested or inspired, so it's easier for reviewers.